### PR TITLE
Add door component

### PR DESCRIPTION
**Pull request by shorowit**
TODO:
- [x] Add door component to UI (similar interface behavior to single windows)
- [x] Add window type dropdown to UI
- [x] Update default_library.json by running create_library.js
  
Addresses #214.

### CODE REVIEW

**Review by bgschiller — APPROVED** [review]
Looks sensible to me 👍

File: schema/geometry_schema.json [hunk 1]
```diff
@@ -996,10 +1037,59 @@
 				"window_spacing",
 				"height",
 				"width",
+				"window_type",
```
Comment (bgschiller) [inline schema/geometry_schema.json:1040]: I notice we now have "window_definition_type", as well as "window_type". In the UI, these show up as "Type" and "Window Type":


Could that be confusing for people? Perhaps we could call the Fixed/Operable distinction something else?

Comment (bgschiller) [reply schema/geometry_schema.json:1040]: change "Type" to "Mode". Add tests to catch this for backwards compatibility.

**Review by gh-23378 — APPROVED** [review]
looks good!

**Pull request merged**